src/main/java/dev/openfeature/sdk/OpenFeatureClient.java

```diff
@@ -116,7 +122,15 @@ private <T> FlagEvaluationDetails<T> evaluateFlag(FlagValueType type, String key
 
         try {
             // openfeatureApi.getProvider() must be called once to maintain a consistent reference
-            provider = openfeatureApi.getProvider(this.domain);
+            FeatureProviderStateManager stateManager = providerAccessor.getProviderStateManager();
+            provider = stateManager.getProvider();
+            ProviderState state = stateManager.getState();
+            if (ProviderState.NOT_READY.equals(state)) {
+                throw new ProviderNotReadyError("provider not yet initialized");
+            }
+            if (ProviderState.FATAL.equals(state)) {
+                throw new FatalError("provider is in an irrecoverable error state");
+            }
```

Comment:
We had some effort in parallel to handle evaluation errors without exceptions in https://github.com/open-feature/java-sdk/pull/1095.

@toddbaert does this also apply to unexpected Provider states?

In general, how do we handle provider state here? If it's now managed by the SDK, does it only forward the evaluation calls to the provider, if it is in `READY` state, or in any state except `NOT_READY` and `FATAL`?

Reply:
Currently we only forward calls if the provider is not in error (or fatal), which differs from the implementation of the `InMemoryProvider`, where they are only forwarded when the provider is `READY`. Is this what we want?

Reply:
> Currently we only forward calls if the provider is not in error (or fatal), which differs from the implementation of the InMemoryProvider, where they are only forwarded when the provider is READY. Is this what we want?

Yes, this is what we want. Some providers may be in an error state, and still work in some sense (they may send error telemetry, or read from an out-of-date cache, or even try to reconnect before returning). The only states that should short-circuit are FATAL and NOT_READY. This is defined in the spec (and implemented in other SDKs):

https://openfeature.dev/specification/sections/flag-evaluation#requirement-177
https://openfeature.dev/specification/sections/flag-evaluation#requirement-178